languages/go/tests/unit_test.go

```diff
@@ -443,3 +443,86 @@ func TestFailingALot(t *testing.T) {
 		}
 	}
 }
+
+func TestPartial(t *testing.T) {
+	var o oso.Oso
+	var err error
+
+	if o, err = oso.NewOso(); err != nil {
+		t.Fatalf("Failed to set up Oso: %v", err)
+	}
+
+	if o.LoadString("f(1); f(x) if x = 1 and x = 2;") != nil {
+		t.Fatalf("Load string failed: %v", err)
+	}
+
+	q, err := o.NewQueryFromRule("f", Variable("x"))
+	if err != nil {
+		t.Fatalf("Failed to get result: %s", err)
+	}
+	q.SetAcceptExpression(true)
```

Comment:
Removing this has no effect on the test b/c the first result doesn't return an expression (it returns `1` after matching `f(1);`) and the second result doesn't return an expression (`x = 1 and x = 2` are incompatible constraints, so no result is returned). Is this section of the test actually testing anything worthwhile?

https://github.com/osohq/oso/blob/75c6f608072ea1bb2e1b6bed42fce64d4b35a780/languages/go/tests/unit_test.go#L455-L476

Reply:
I copied this test from the python tests 

My assumption was that this is testing that setting allow expression doesn't suddenly break everything else?

Reply:
Okay I suppose that's fine. Would also be nice to add a test that _doesn't_ set the flag and asserts we receive the error — maybe even combine it with the final test case here but modify the policy to produce a pair of expression results, and we frob the flag in between the results and see one of them error & the other return the `Expression`.

Reply:
Okay, made the test case reusable, and checked that it works for:
- not setting
- setting to false
- setting to true